node/node_v2.go

```diff
@@ -0,0 +1,126 @@
+// These v2 methods are implemented in this separate file to keep the code organized.
+// Note that there is no NodeV2 type and these methods are implemented in the existing Node type.
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+	"github.com/Layr-Labs/eigenda/core"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/gammazero/workerpool"
+)
+
+type requestMetadata struct {
+	blobShardIndex int
+	quorum         core.QuorumID
+}
+type relayRequest struct {
+	chunkRequests []*clients.ChunkRequestByRange
+	metadata      []*requestMetadata
+}
+type response struct {
+	metadata []*requestMetadata
+	bundles  [][]byte
+}
+
+type RawBundles struct {
+	BlobCertificate *corev2.BlobCertificate
+	Bundles         map[core.QuorumID][]byte
+}
+
+func (n *Node) DownloadBatch(ctx context.Context, batch *corev2.Batch) ([]*corev2.BlobShard, []*RawBundles, error) {
+	if n.RelayClient == nil {
+		return nil, nil, fmt.Errorf("relay client is not set")
+	}
+
+	operatorState, err := n.ChainState.GetOperatorStateByOperator(ctx, uint(batch.BatchHeader.ReferenceBlockNumber), n.Config.ID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	blobShards := make([]*corev2.BlobShard, len(batch.BlobCertificates))
+	rawBundles := make([]*RawBundles, len(batch.BlobCertificates))
+	requests := make(map[corev2.RelayKey]*relayRequest)
+	for i, cert := range batch.BlobCertificates {
+		blobKey, err := cert.BlobHeader.BlobKey()
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to get blob key: %v", err)
+		}
+
+		if len(cert.RelayKeys) == 0 {
+			return nil, nil, fmt.Errorf("no relay keys in the certificate")
+		}
+		blobShards[i] = &corev2.BlobShard{
+			BlobCertificate: cert,
+			Bundles:         make(map[core.QuorumID]core.Bundle),
+		}
+		rawBundles[i] = &RawBundles{
```

Comment:
It seems like it could be cleaner to do something like 

```
type BlobShardAndRawBundles struct{
    core.BlobShard
    Bundles: map[core.QuorumID][]byte
}
```

and just return one []BlobShardAndRawBundles array?

Reply:
Or maybe the []core.BlobShard array is needed by another interface.

Reply:
It's useful to keep the return type `[]*corev2.BlobShard` and `[]*RawBundles` (separate slices for each type of bundles) because you want the raw bundles for storage and blob shards for validation. If combined, we'd end up reconstructing these slices downstream